subprojects/base-services/src/main/java/org/gradle/internal/time/Time.java

```diff
@@ -20,29 +20,29 @@
 
 public abstract class Time {
 
-    private static final Clock CLOCK = new MonotonicClock();
+    private static final ClockSync CLOCK_SYNC = new DefaultClockSync(TimeSource.SYSTEM);
```

Comment:
Could we get rid of the mutable static and instead have an injectable `Clock` which is bound to the build scope? That would a lot of the added ceremony like `ClockSync` and `ClockSyncingBuildActionExecutor` etc.

Reply:
Ill give it a try. Previously this wasn't possible because this clock was more widely used.

Reply:
I've done that and it appears to work. However, @adammurdoch is now pushing in the original direction.